src/main/java/se/citerus/dddsample/interfaces/booking/web/CargoAdminController.java

```diff
@@ -39,7 +40,8 @@
 @RequestMapping("/admin")
 public final class CargoAdminController {
 
-    private final BookingServiceFacade bookingServiceFacade;
+    @Autowired
```

Comment:
Can this dependency be injected into the constructor instead?

Reply:
I think  it's hard and dangerous to do it since the injection currently has some circular.
It's the next step I'm planning.

Reply:
Okay, got it. In that case, could you add a comment to this field explaining that it cannot be moved into the constructor params due to the circular dependency problem? Otherwise, my knee-jerk reaction seeing that would be to inline it 😃

Reply:
I got the reason by new investigation and make it inline with constructor

Reply:
May I know why some beans are defined in the context file (like DDDSampleApplicationContext) but some are defined by the annotation?

I think the domain layer should not have the runtime knowledge like CDI, but the infrastructure and interfaces may can do it. 

I have some idea for optimizing the  bean definitions to:
1. no  circular reference
2. no duplicate bean definition

But I'm wondering about how to define the beans with the original principle.

Reply:
Looking at the code, I can't find any beans that are defined outside the context files. Can you give an example of a bean you found that is declared outside of a context file?

Reply:
see HandlingReportServiceImpl this class is defined outside the context files.
and TrackCommandValidator this class is defined inside the context file but CargoTrackingController is using it by new .

Reply:
Right, CargoTrackingController should inject TrackCommandValidator instead of instantiating by new. 

As for HandlingReportServiceImpl, using the @RestController annotation leads to significantly less configuration code, so I wouldn't want to refactor that to be instantiated programmatically in the context class.